### PR TITLE
Prompt user for confirmation before allowing them to change their own role & update webstate if they do

### PR DESCRIPTION
**Pull request by johnpcooke94**
This PR will prompt the user for confirmation if they attempt to alter their own role in the admin panel. The user will have the option to cancel this action if they read the warning and decide they no longer wish to do so.

In addition, the webstate should be updated so that the user loses access to the admin panel if they do choose to change their own role from admin to another role.

### CODE REVIEW

File: src/components/AdminPanelLayout/UsersPane.tsx [hunk 1]
```diff
@@ -1,42 +1,156 @@
...
+        </tbody>
+      </Table>
+    </div>
+  )}
+  </Layout>;
 
 };
 
-export default UsersPane;
+export default connect((state: WebState) => ({currentUser: getCurrentUser(state) }))(UsersPane);
```
Comment (johnpcooke94) [inline src/components/AdminPanelLayout/UsersPane.tsx]: I'm having a hard time getting the webstate to update with the response data from here. I think I'm missing something with how this redux function works.

Comment (johnpcooke94) [reply src/components/AdminPanelLayout/UsersPane.tsx]: I found a solution to get it working. Resolving this.

File: src/components/AdminPanelLayout/UsersPane.tsx [hunk 2]
```diff
@@ -1,57 +1,157 @@
...
+          <ModalFooter>
+            <Button variant={'primary'} onClick={handleCancel}>Cancel</Button>
+            <Button variant={'secondary'} onClick={handleConfirm}>Confirm</Button>
+          </ModalFooter>
+        </Modal>
+      </Layout>
+    );
+  };
+
+  const CommitStatus = () => {
```
Comment (johnpcooke94) [inline src/components/AdminPanelLayout/UsersPane.tsx:127]: @jasekiw Would you rather inline this component? I thought it was pretty messy-looking with that nested ternary operator (and sonarcloud agreed) so I separated it out into this component which seems a lot more readable to me. Let me know if you prefer it inline though and I can change it back.

**Review by jasekiw — APPROVED** [review]

**Comment by jasekiw** [comment]
@johnpcooke94 Merged.

**Pull request merged**